crates/completest-pty/src/lib.rs

```diff
@@ -443,6 +445,126 @@ impl Runtime for ElvishRuntime {
     }
 }
 
+/// Abstract factory for [`PowershellRuntime`]
+#[derive(Debug)]
+#[non_exhaustive]
+pub struct PowershellRuntimeBuilder {}
+
+impl RuntimeBuilder for PowershellRuntimeBuilder {
+    type Runtime = PowershellRuntime;
+
+    fn name() -> &'static str {
+        "powershell"
+    }
+
+    fn new(bin_root: PathBuf, home: PathBuf) -> std::io::Result<Self::Runtime> {
+        PowershellRuntime::new(bin_root, home)
+    }
+
+    fn with_home(bin_root: PathBuf, home: PathBuf) -> std::io::Result<Self::Runtime> {
+        PowershellRuntime::with_home(bin_root, home)
+    }
+}
+
+/// Powershel runtime
+#[derive(Debug)]
+#[cfg(unix)] // purely for rustdoc to pick it up
+pub struct PowershellRuntime {
+    path: OsString,
+    home: PathBuf,
+    config: PathBuf,
+    timeout: Duration,
+}
+
+impl PowershellRuntime {
+    /// Initialize a new runtime's home
+    pub fn new(bin_root: PathBuf, home: PathBuf) -> std::io::Result<Self> {
+        std::fs::create_dir_all(&home)?;
+
+        let config_path = home.join("powershell/Microsoft.PowerShell_profile.ps1");
+        let config = "function prompt {
+    '% '
+}
+Set-PSReadLineOption -PredictionSource None
+"
+        .to_owned();
+        std::fs::create_dir_all(config_path.parent().expect("path created with parent"))?;
+        std::fs::write(config_path, config)?;
+
+        Self::with_home(bin_root, home)
+    }
+
+    /// Reuse an existing runtime's home
+    pub fn with_home(bin_root: PathBuf, home: PathBuf) -> std::io::Result<Self> {
+        let config_path = home.join("powershell/Microsoft.PowerShell_profile.ps1");
+        let path = build_path(bin_root);
+
+        Ok(Self {
+            path,
+            home,
+            config: config_path,
+            timeout: Duration::from_millis(50),
```

Comment:
Setting the timeout is 50ms will get nothing in `comptest` function.
Because https://github.com/assert-rs/completest/blob/4334bff3a145b1ab4c18451909b67217a270ca22/crates/completest-pty/src/lib.rs#L495
the read call is slow for powershell.